lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1857,7 +1857,7 @@ def _resolve_depenency_map(
         parents = [
             "%s.%s:%s" % (p.namespace, p.name, p.ver)
             for p in dep_exc.criterion.iter_parent()
-            if p is not None
+            if p is not None and not p.is_virtual
```

Comment:
Let's try this:
```suggestion
            str(p) for p in dep_exc.criterion.iter_parent()
            if p is not None
```
See https://github.com/ansible/ansible/blob/65a96daa/lib/ansible/galaxy/dependency_resolution/dataclasses.py#L464-L473

Reply:
cleaner